src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -21,18 +22,32 @@ class MessageNotifier(
   @Value("\${aws.sns.topic_arn}")
   private val topicArn: String
 ) {
-  fun send(hearingEvent: HearingEvent) {
+  fun send(telemetryEventType: TelemetryEventType, hearingEvent: HearingEvent) {
 
-    val messageValue = MessageAttributeValue()
+    val messageTypeValue = MessageAttributeValue()
       .withDataType("String")
       .withStringValue(MESSAGE_TYPE)
```

Comment:
Confused with this actually, If CHER not sending the correct message type how does the matcher know where the case (hearing) coming from? 🤔

Reply:
I say it's not sending the correct message type but that's not strictly true - what I suppose I mean is as part of this change we need it to be more specific. Currently it's mapped [here](https://github.com/ministryofjustice/court-case-matcher/blob/main/src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/model/MessageType.java) and all CHER messages have type `CP_TEST_COURT_CASE`. This was supposed to be  a temporary name that until now we've never really had any reason to change. It was always the intention that we'd differentiate between `CONFIRMED_UPDATED` and `RESULTED` types though and now we need to make that distinction.

Reply:
Agree on the message types but my confusion is this. If all messages have the `CP_TEST_COURT_CASE` as a type how this code works?

```    Hearing extractHearing(String payload, String messageId) {
        try {
            SnsMessageContainer snsMessageContainer = snsMessageWrapperJsonParser.parseMessage(payload, SnsMessageContainer.class);
            log.debug("Extracted message ID {} from SNS message of type {}. Incoming message ID was {} ", snsMessageContainer.getMessageId(), snsMessageContainer.getMessageType(), messageId);

            return switch (snsMessageContainer.getMessageType()){
                case LIBRA_COURT_CASE -> libraParser.parseMessage(snsMessageContainer.getMessage(), LibraHearing.class).asDomain();
                case COMMON_PLATFORM_HEARING -> parseCPMessage(snsMessageContainer);
                default -> throw new IllegalStateException("Unprocessable message type: " + snsMessageContainer.getMessageType());
            };

        }